### ISSUE TITLE
signal SIGABRT when trying to open database

### ISSUE THREAD

**Issue by PaulMest** [craftzdog/react-native-sqlite-2#56]
My team recently updated a React Native application to 0.59 and found that our previous PouchDB configuration with SQLite and Cordova adpaters would no longer work. Thanks to your libraries it is now working in general 🎉, but runs into a weird crash that is hard to pin down.

It seems like when we try to load many PouchDB databases (~100 databases), the application will crash with error `signal SIGABRT`.

```obj-c
// File: RNSqlite2.m

-(NSValue*)openDatabase: (NSString*)dbName {
  logDebug(@"opening DB: %@", dbName);
  NSValue *cachedDB = [cachedDatabases objectForKey:dbName];
  if (cachedDB == nil) {
    logDebug(@"opening new db");
    NSString *fullDbPath = [self getPathForDB: dbName];
    logDebug(@"full path: %@", fullDbPath);
    const char *sqliteName = [fullDbPath UTF8String];
    sqlite3 *db;
    if (sqlite3_open(sqliteName, &db) != SQLITE_OK) {
      logDebug(@"cannot open database: %@", dbName); // shouldn't happen
    };
    cachedDB = [NSValue valueWithPointer:db];

    // **** Thread 499: signal SIGABRT on this line...
    [cachedDatabases setObject: cachedDB forKey: dbName];

  } else {
    logDebug(@"re-using existing db");
  }
  return cachedDB;
}
```

Screen Shot:
![image](https://user-images.githubusercontent.com/1721104/57432422-b4232980-71ea-11e9-84ce-0681e516c79d.png)

Any idea what could be the issue? Or what we could do to help debug this?

**Comment by craftzdog** [comment]
Hi, 

I guess there may be a multithreading issue.
Can you try it with `@synchronized` to synchronize the method?

**Comment by PaulMest** [comment]
Thanks @craftzdog. This appears to have worked! We're going to test for the remaining of the day. If we do not find any more issues, I'll submit a PR for your review.

**Comment by craftzdog** [comment]
Looks great. Waiting for your PR, thanks!

**Comment by PaulMest** [comment]
This was fixed as part of #58

**Comment by craftzdog** [comment]
Released v2.0.2. Thanks again!
